notification-service/src/constants/SUBSCRIPTIONS.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import { EventSubscription } from '../types';
+import { ETHERSCAN_API_KEY, ETHEREUM_NETWORK } from '../variables';
+import { getEtherscanAPI, getEtherscanURL } from './NETWORKS';
+import { ethers } from 'ethers';
+
+export async function getAbiFromContractAddress(contract: string) {
+    if (!ETHERSCAN_API_KEY) {
+        throw new Error('subscriptions: please set a valid ETHERSCAN_API_KEY, in the environment files');
+    }
```

Comment:
Does it mean, it will only fail at runtime if no `ETHERSCAN_API_KEY` provided?

Reply:
I changed this, to verify it before hand!